gcs/conn.go

```diff
@@ -183,8 +183,8 @@ func (c *conn) OpenBucket(
 	// Forbidden". Similarly for bad bucket names that don't collide with another
 	// bucket.
 	_, err = b.ListObjects(ctx, &ListObjectsRequest{MaxResults: 1})
-
-	if typed, ok := err.(*googleapi.Error); ok {
+	unwrappedErr := errors.Unwrap(err)
```

Comment:
Directly calling Unwrap is not very clear — it doesn’t show which error type you’re interested in, and it assumes that precisely one level of wrapping is happening.

Can you use `errors.As` instead? See https://go.dev/blog/go1.13-errors for more details.

Reply:
You’re still calling `errors.Unwrap`, and you shouldn’t.

You also don’t need to check `if err != nil`, because `errors.As()` will only return true on non-nil errors.

Try this:
```go
	var apiError *googleapi.Error
	if errors.As(err, &apiError) {
		switch apiError.Code {
// …
```

Reply:
This error type is "wrapError" which has below type of structure
type wrapError struct {
	msg string
	err error
}
func (e *wrapError) Error() string {
	return e.msg
}
func (e *wrapError) Unwrap() error {
	return e.err
}
so without unwrapping it we can't get error code or anything else. the above code which you are talking about is returning nil.

Reply:
No: `errors.As` calls `Unwrap` under the hood for you when required: https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/errors/wrap.go;l=99;drc=2580d0e08d5e9f979b943758d3c49877fb2324cb

Reply:
Yes,thanks Fixed it